README.md

```diff
@@ -1 +1,997 @@
+authorization checks using Ballerina.
+
+The following are the sections available in this guide.
+
+- [What you'll build](#what-youll-build)
+- [Prerequisites](#prerequisites)
+- [Developing the service](#developing-the-service)
+- [Testing](#testing)
+- [Deployment](#deployment)
+- [Observability](#observability)
+
+## What you’ll build
+To understand how you can secure a RESTful web service using Ballerina, let’s secure the web service you created in
+the [RESTful Service guide](https://ballerina.io/learn/by-guide/restful-service/).
+
+The following figure illustrates all the functionalities of the OrderMgt RESTful web service, that should be secured.
+
+![RESTful Service](images/restful-service.svg "RESTful Service")
+
+- **Create Order** : Order creation should be allowed for all the users. Authentication or authorization should not be enforced for this function.
+- **Retrieve Order** : Retrieval of the order details should only be allowed for authenticated users.
+- **Update Order** : Updating order details should only be allowed for "admin" users.
+- **Delete Order** : Deletion of the order should only be allowed for "admin" users.
+
+## Prerequisites
+
+- JDK 1.8 or later
+- [Ballerina Distribution](https://github.com/ballerina-lang/ballerina/blob/master/docs/quick-tour.md)
+- A Text Editor or an IDE
+- [Docker](https://docs.docker.com/engine/installation/)
+- To provide required LDAP user store for this guide, we have created a docker image with an embedded Apache DS which contains the user details that should be used in authentication and authorization. we will use the following LDAP schema, which creates two users. The 'counter' user only has 'add_order' scope, whereas the 'admin' user has 'add_order', 'update_order' and 'cancel_order' scopes. Here scopes are equivalent to LDAP groups. Follow the intruction to setup LDAP server neccessory for the guide.
+- Execute the following command to pull docker image from the docker hub.
+```bash
+docker pull vijithaekanayake/embedded_ldap_server:1.0.0
+```
+- After pulling the docker image, execute below command to start the LDAP server.
+```bash
+docker run -d -p 9389:9389 vijithaekanayake/embedded_ldap_server:1.0.0
+```
+
+### Optional requirements
+- Ballerina IDE plugins ([IntelliJ IDEA](https://plugins.jetbrains.com/plugin/9520-ballerina), [VSCode](https://marketplace.visualstudio.com/items?itemName=WSO2.Ballerina), [Atom](https://atom.io/packages/language-ballerina))
+
+## Developing the service
+
+- We can get started with a Ballerina service; 'OrderMgtService', which is the RESTful service that serves the order management request. We will look at securing multiple resources exposed by OrderMgtService to match with the different security requirements.
+
+- Although the language allows you to have any package structure, use the following package structure for this project to follow this guide. The root directory will be denoted by `SAMPLE_ROOT` in the context of this README.
```

Comment:
```suggestion
- Although the language allows you to have any package structure, use the following package structure for this project to follow this guide. The root directory is denoted by `SAMPLE_ROOT` in the context of this README.
```

Let's use simple present tense instead of future tense wherever possible.